src/env.c

```diff
@@ -25,7 +25,7 @@ extern "C"
 #include "rcutils/error_handling.h"
 
 bool
-rcutils_set_env(const char * env_name, const char * env_value)
+rcutils_set_env_overwrite(const char * env_name, const char * env_value, bool overwrite)
```

Comment:
`_putenv_s` does not have overwrite argument, so probably we need to call `getenv` to check if the environmental variable exists or not?

Reply:
Yes, Windows annoyingly doesn't have this.  I ran into it when implementing the equivalent for Gazebo.  Generally, I think you can assume that as long as the arguments are valid, then the overwrite will succeed.

In this case, I think there are also some logic bugs.

For Windows, if the env is already set and you call `set_env_overwrite` with `overwrite=false`, you will end up overwriting (silently).

Reply:
Thanks for highlighting these points. I guess the user should rely on `rcutils_get_evn` to check if an envar is set and only then call `rcutils_set_env`? If so we can close this PR.